src/states/Postroll.js

```diff
@@ -0,0 +1,139 @@
+import videojs from 'video.js';
+
+import {AdState, BeforePreroll, Preroll, AdsDone} from '../states.js';
+import {startAdBreak, endAdBreak} from '../adBreak.js';
+
+export default class Postroll extends AdState {
+
+  init(player) {
+    // Legacy name that now simply means "handling postrolls".
+    player.ads._contentEnding = true;
+
+    // Start postroll process.
+    if (!player.ads.nopostroll_) {
+      player.addClass('vjs-ad-loading');
+
+      this._postrollTimeout = player.setTimeout(function() {
+        player.trigger('adtimeout');
+      }, player.ads.settings.postrollTimeout);
+
+    // No postroll, ads are done
+    } else {
+      player.setTimeout(() => {
+        player.ads.debug('Triggered ended event (no postroll)');
+        this.contentResuming = true;
+        player.trigger('ended');
+      }, 1);
+    }
+  }
+
+  onAdsError(player) {
+    // In the future, we may not want to do this automatically.
+    // Integrations should be able to choose to continue the ad break
+    // if there was an error.
+    if (player.ads.inAdBreak()) {
+      player.ads.endLinearAdMode();
+    }
+  }
+
+  /*
+   * Start the postroll if it's not too late.
+   */
+  startLinearAdMode() {
+    const player = this.player;
+
+    if (!player.ads.isAdPlaying() && !this.contentResuming) {
+      player.ads.debug('startLinearAdMode (Postroll)');
+      player.ads.adType = 'postroll';
+      player.clearTimeout(this._postrollTimeout);
+      startAdBreak(player);
+    } else {
+      videojs.log.warn('Unexpected startLinearAdMode invocation (Postroll)');
+    }
+  }
+
+  /*
+   *
+   */
+  onAdStarted(player) {
+    player.removeClass('vjs-ad-loading');
```

Comment:
can this go in the AdState class?

Reply:
`vjs-ad-loading` management is very different for prerolls and postrolls, so to keep it as clear as possible I isolated it to those classes rather than trying to share code.

Reply:
I am updating the empty comment here though.